### PR TITLE
bitbucket_pipeline_variable: Change pagination logic

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Bitbucket's pagination for pipeline variables is flawed.
Refactor bitbucket_pipeline_variable code to correct this logic.

Fixes: #1425

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/1425_bitbucket_pipeline_variable.yml
plugins/modules/source_control/bitbucket/bitbucket_pipeline_variable.py

### CODE REVIEW

File: plugins/modules/source_control/bitbucket/bitbucket_pipeline_variable.py [hunk 1]
```diff
@@ -119,17 +119,16 @@ def get_existing_pipeline_variable(module, bitbucket):
...
-    }
-
+    variables_base_url = BITBUCKET_API_ENDPOINTS['pipeline-variable-list'].format(
+        username=module.params['username'],
+        repo_slug=module.params['repository'],
+    )
     # Look through the all response pages in search of variable we need
-    while 'next' in content:
+    page = 1
+    while page:
```
Comment (felixfontein) [inline plugins/modules/source_control/bitbucket/bitbucket_pipeline_variable.py:128]:
```suggestion
    while True:
```
If you're creating an infinite loop, be honest about it :)

Comment (Akasurde) [reply plugins/modules/source_control/bitbucket/bitbucket_pipeline_variable.py:128]: 😀

**Review by felixfontein — APPROVED** [review]
Code looks good. Cannot test whether this works though, someone else has to do that.

**Comment by Akasurde** [comment]
@felixfontein @rik2803 already tested code - https://github.com/ansible-collections/community.general/issues/1425#issuecomment-746445505

**Comment by felixfontein** [comment]
@Akasurde thanks for fixing this!

**Pull request merged**